src/mapserv-index.cpp

```diff
@@ -374,7 +374,11 @@ static mapObj *getMapFromConfig(configObj *config, const char *key) {
   char pathBuf[MS_MAXPATHLEN];
   mapfilePath = msConfigGetMap(config, key, pathBuf);
 
-  return msLoadMap(mapfilePath, nullptr, config);
+  char *mapfilePathCopy = msStrdup(mapfilePath);
```

Comment:
mapfilePath should be tested against NULL. The current code will not crash, but msStrdup(nullptr) returns an empty string, and that will likely lead to a confusing error message in msLoadMap. If catching NULL before, we can emit a message like `` "No map associated with key %s", key ``